### ISSUE TITLE
ESR Only FIXED in TB154 -  [Webextension Regression] - Attachment "Save all" applies only to orginal attachments

### ISSUE THREAD

**Issue by dugite-code** [TB-throwback/LookOut-fix-version#105]
This was a regression introduced with the migration to webextension. When selecting the Save All button for attachments only the winmail.dat and other original attachments are saved. Users currently must drag and drop files or `open the containing folder` to access files.

**Comment by jobisoft** [comment]
I think I will fix this in core directly, together with the work to add the Attachments API used by lookout fixed. ETA: 1 month.

**Comment by nckramer** [comment]
thank you. and apologies for the duplicate. the "open" is a good workaround.

**Comment by dugite-code** [comment]
@jobisoft has there been any changes upstream so far with this?

**Comment by jobisoft** [comment]
It is on track for the next ESR.

**Comment by AndyM48** [comment]
Is this the same error as - highlight file > select Save As > locate folder and Save - nothing is saved?

**Comment by nckramer** [comment]
Yes. The temporary workaround is use "open" or "open all" when prompted what to do with the files, choose "save", pick a location, and check the box for always do the same thing.

You still get the same result. The files are extracted & saved. Just the method is a little different.

**Comment by davidmorr** [comment]
I think I this problem still exists, but perhaps in a slightly different form to the original bug above.

macOS 12.7.6 on Mac Mini M1, Thunderbird 115.18.0 (64-bit)

If I double-click the attachment name at the bottom of the message window, the attachment opens in an appropriate application (Preview for PDF, TextEdit for RTF, etc). I can then select Save As... in the app and save a full copy.

If I select Show in Finder, the full file is in this folder:
/private/var/folders/vx/cd13wgwd5_nd24r4jl2xp2480000gn/T 

Clicking Save All button does nothing at all afaics.

(Right click)|Save As... on an attachment puts up a file dialogue with the correct name, but clicking Open leaves nothing in the selected folder.

Click and drag the attachment name will create a file where it is dragged to, but it is zero length. (This is the main way many Mac users extract attachments to save in specific locations. It is particularly treacherous that it seems to create a file which may leave people thinking they have saved a copy of the attachment.)
So while it is possible to get the attachment, most methods people are likely to want to use do not work.

I found this in the Error Console:
[LookOut errors.txt](https://github.com/user-attachments/files/18385947/LookOut.errors.txt)

**Comment by nckramer** [comment]
The only workaround, although un-intuitive is to "open all". (same spot as "save all") There is no dialog box, save box etc. It just dumps the attached files to your desktop or designated "save to" spot.
All other ways to save attachments out of these Microsoft emails is still broken.

**Comment by davidmorr** [comment]
Open All just opens them all, same as double-clicking on the individual filenames. It does not leave them in Downloads or on the desktop.

**Comment by nckramer** [comment]
@davidmorr - it can save all files instead of opening them. it's quicker & less hassle. but yes, it needs setup a certain way.
- choose "save file" instead of "open with", and "do this automatically"
    once you've done this with all the file types you get, when you select "open all" it will just save all the files to the designated location. or you can set your preferences in settings, or change it  back to "always ask" if you don't want to automatically save them anymore.
![Screen-Shot-2025-01-12-at-1 03 36-PM](https://github.com/user-attachments/assets/989fe6e1-3372-47c4-a6b5-c9106febc69c)
![Screen-Shot-2025-01-12-at-1 01 26-PM](https://github.com/user-attachments/assets/c419c462-1314-4ace-9465-5b79572e1fda)

**Comment by dugite-code** [comment]
Sorry I've been unresponsive

Thanks @nckramer that's an excellent writ-up on the workaround.

@davidmorr as mentioned by @jobisoft above we are currently waiting on an upstream change in the ESR release.

**Comment by AndyM48** [comment]
I wrote a simple script to unpack the attachment to a temporary folder and open that folder in my file manager. From there it is easy, of course, to open a file,  copy, save, saveall etc. All that was needed was to associate the script to 'application/ms-tnef' in thunderbird.

**Comment by dugite-code** [comment]
Note This is fixed in TB 154
